routes/admin/users.rb

```diff
@@ -158,6 +158,34 @@
     }
 end
 
+post '/admin/users/:user_id/renew/?' do
+    if params[:user_id].to_i == @user.id
+        user = @user
+    else
+        user = User.includes(:permissions).where(:id => params[:user_id], :service_space_id => SS_ID).first
+    end
+
+    if user.nil?
+        flash :alert, "Not Found", "Sorry, that user was not found"
+        redirect '/admin/users/'
+    end
+
+    user.update({
+        :expiration_date => Date.today + 30
+    })
+
+    status = "expired"
```

Comment:
Make sure to preserve the user's opt-out status when setting their space_status. There is logic for this in auth.rb